File: tests/fixtures/ruby-simple/lib/utils/logger.rb
# Simple logging utility.
module Utils
  class Logger
    attr_reader :context

    def initialize(context)
      @context = context
    end

    def info(message)
      puts "[INFO] #{@context}: #{message}"
    end

    def error(message)
      $stderr.puts "[ERROR] #{@context}: #{message}"
    end

    def debug(message)
      puts "[DEBUG] #{@context}: #{message}"
    end
  end
end

File: tests/fixtures/swift-simple/Sources/Utils/Logger.swift
import Foundation

/// Simple logging utility.
class Logger {
    private let context: String

    init(context: String) {
        self.context = context
    }

    func info(_ message: String) {
        print("[INFO] \(context): \(message)")
    }

    func error(_ message: String) {
        print("[ERROR] \(context): \(message)")
    }

    func debug(_ message: String) {
        print("[DEBUG] \(context): \(message)")
    }
}
